### PR TITLE
move gke nodes firewall rule from vpc to gke

### PR DESCRIPTION
**Pull request by eduartua**
This solves issue #7

### CODE REVIEW

File: modules/gcp/gke/main.tf [hunk 1]
```diff
@@ -149,4 +149,20 @@ resource "google_project_iam_member" "service_account-roles" {
   project = var.gcp["project"]
   role    = each.value
   member  = "serviceAccount:${google_service_account.gke_service_account.email}"
-}
\ No newline at end of file
+}
+
+resource "google_compute_firewall" "cert-manager-webhook-firewall" {
+  name = "${var.region}-${var.environment}-cert-manager-webhook-firewall-rule"
+  network = var.network
```
Comment (victortrac) [inline modules/gcp/gke/main.tf]: Is there some way to have this use the module self_link for network if var.network isn't defined?

Comment (eduartua) [reply modules/gcp/gke/main.tf]: Is this to make the `gke` module independent of the `vpc` network output? When using the `vpc` module, we are already using the self_link for network:

https://github.com/cloudkite-io/terraform-modules/blob/master/modules/gcp/vpc/outputs.tf#L3

In the case of changing this `var.network` to use module self_link for network the only option is data source, although we will still need the network name.

Comment (victortrac) [reply modules/gcp/gke/main.tf]: We don't need to use the data source if the name of the self_link is deterministic based on the name value input given to the network module. We generate the name for the network, so there's no reason we can't guess the name of the network if it's not specified.

Comment (eduartua) [reply modules/gcp/gke/main.tf]: Changes already tested in dev and pushed.

File: modules/gcp/gke/variables.tf [hunk 2]
```diff
@@ -1,3 +1,8 @@
+variable "project" {
+  description = "The project ID to host the cluster in (required)"
+  type = "string"
+}
+
```
Comment (victortrac) [inline modules/gcp/gke/variables.tf]: https://github.com/cloudkite-io/terraform-modules/blob/master/modules/gcp/gke/main.tf#L2

We already use the project in a var. Either use that value and make it explicitly called out as a required map in the README or switch all references to this. But don't do both.

File: modules/gcp/gke/variables.tf [hunk 3]
```diff
@@ -13,16 +18,27 @@ variable "gke" {
   type = "map"
 }
 
+variable "region" {
+  description = "The region to host the cluster in"
+  type        = string
+}
```
Comment (victortrac) [inline modules/gcp/gke/variables.tf]: Again. This is already passed in.

Comment (eduartua) [reply modules/gcp/gke/variables.tf]: Already fixed.

**Pull request merged**